File: back-end/src/main/java/server/model/KatedraNastavnik.java
package server.model;

import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.ManyToOne;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

@Entity
@Data
@AllArgsConstructor
@NoArgsConstructor
public class KatedraNastavnik {
	
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @ManyToOne(optional = false)
    private Katedra katedra;

    @ManyToOne(optional = false)
    private Nastavnik nastavnik;
}

File: front-end/src/app/models/KatedraNastavnik.ts
import { Katedra } from "./Katedra";
import { Nastavnik } from "./Nastavnik";


export interface KatedraNastavnik {
  id?: number | null;
  vidljiv: boolean;
  katedra: Katedra;      
  nastavnik: Nastavnik;  
}
